s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/DeleteImportCandidatesEventEmitter.java

```diff
@@ -49,56 +48,46 @@ public static EventBridgeClient defaultEventBridgeClient() {
-            .forEach(this::logWarningForNotEmittedEvents);
+        var events = extractIdentifiersFromEventBlob(input)
+                    .map(this::createEvent)
+                    .toList();
+        var putEventsResults = emitEvents(events, context);
+        logWarningForNotEmittedEvents(putEventsResults);
         return null;
     }
 
-    private static List<String> splitOnNewLine(String string) {
-        return Arrays.asList(string.split("\n"));
-    }
-
-    private static boolean isEmptyLine(String string) {
+    private static boolean isNotEmptyLine(String string) {
         return !string.isBlank() || !string.isEmpty();
     }
 
     private ImportCandidateDeleteEvent createEvent(String id) {
         return new ImportCandidateDeleteEvent(ImportCandidateDeleteEvent.EVENT_TOPIC, id);
     }
 
-    private List<ImportCandidateDeleteEvent> createEvents(Stream<String> scopusIdentifiers) {
-        return scopusIdentifiers.map(this::createEvent)
-                   .collect(Collectors.toList());
-    }
-
     private List<PutEventsResult> emitEvents(List<ImportCandidateDeleteEvent> events, Context context) {
-        var batchEventEmitter = new BatchEventEmitter<ImportCandidateDeleteEvent>(
-            ImportCandidateDeleteEvent.class.getCanonicalName(),
-            context.getInvokedFunctionArn(),
-            eventBridgeClient);
-        logger.info("Events to emit: {}",
-                    events.stream().map(ImportCandidateDeleteEvent::toJsonString).collect(Collectors.toList()));
+        var batchEventEmitter = getBatchEventEmitter(context);
+        logger.info("Events to emit: {}", events.stream().map(ImportCandidateDeleteEvent::toJsonString).toList());
         batchEventEmitter.addEvents(events);
         return batchEventEmitter.emitEvents(NUMBER_OF_EMITTED_ENTRIES_PER_BATCH);
     }
 
-    private Stream<String> extractIdentifiers(String string) {
-        return splitOnNewLine(string).stream()
-                   .filter(DeleteImportCandidatesEventEmitter::isEmptyLine)
-                   .map(this::extractScopusIdentifier);
+    private BatchEventEmitter<ImportCandidateDeleteEvent> getBatchEventEmitter(Context context) {
+        return new BatchEventEmitter<>(ImportCandidateDeleteEvent.class.getCanonicalName(),
+                                       context.getInvokedFunctionArn(),
+                                       eventBridgeClient);
     }
 
     private String extractScopusIdentifier(String item) {
         return item.split(SCOPUS_IDENTIFIER_DELIMITER)[1];
     }
 
-    private String readFile(S3Event event) {
+    private Stream<String> extractIdentifiersFromEventBlob(S3Event event) {
         var s3Driver = new S3Driver(s3Client, extractBucketName(event));
         var fileUri = createS3BucketUri(event);
-        return s3Driver.getFile(UriWrapper.fromUri(fileUri).toS3bucketPath());
+        return s3Driver.getFile(UriWrapper.fromUri(fileUri).toS3bucketPath())
+                   .lines()
```

Comment:
`String::lines` is a simpler method to use than manual attempts to split lines since errors (\r\n) creep in